python/src/scipp/compat/mantid.py

```diff
@@ -546,3 +582,153 @@ def load(filename="",
+    """
+    Use the values provided to create a Mantid workspace.
+
+    The Mantid layout expect the spectra to be the Outer-most dimension,
+    i.e. y.shape[0]. If that is not the case you might have to transpose
+    your data to fit that, otherwise it will not be aligned correctly in the
+    Mantid workspace.
+
+    :param x: Data to be used as X for the Mantid workspace.
+    :param y: Data to be used as Y for the Mantid workspace.
+    :param e: Data to be used as error for the Mantid workspace.
+              If `None` the np.sqrt of y will be used.
+    :param coord_dim: Dim of the coordinate, to be set as the equivalent
+                      UnitX on the Mantid workspace.
+    :param instrument_file: Instrument file that will be
+                            loaded into the workspace
+    :returns: Workspace2D containing the data for X, Y and E
+    """
+    try:
+        import mantid.simpleapi as mantid
+    except ImportError:
+        raise ImportError(
+            "Mantid Python API was not found, please install Mantid framework "
+            "as detailed in the installation instructions (https://scipp."
+            "readthedocs.io/en/latest/getting-started/installation.html)")
+
+    assert len(y.shape) == 2, "Currently can only handle 2D data."
+
+    e = e if e is not None else np.sqrt(y)
+
+    unitX = validate_dim_and_get_mantid_string(coord_dim)
+
+    nspec = y.shape[0]
+    nbins = x.shape[1]
+    nitems = y.shape[1]
+
+    ws = mantid.WorkspaceFactory.create("Workspace2D",
+                                        NVectors=nspec,
+                                        XLength=nbins,
+                                        YLength=nitems)
+
+    for i in range(nspec):
+        ws.setX(i, x[i])
+        ws.setY(i, y[i])
+        ws.setE(i, e[i])
+
+    # Set X-Axis unit
+    ws.getAxis(0).setUnit(unitX)
+
+    if instrument_file is not None:
+        mantid.LoadInstrument(ws,
+                              FileName=instrument_file,
+                              RewriteSpectraMap=True)
+
+    return ws
+
+
+def fit(ws, function, workspace_index, start_x, end_x):
+    """
+    Performs a fit on the workspace.
+
+    :param ws: The workspace on which the fit will be performed
+    :param function: The function used for the fit. This is anything
+                     that mantid.Fit's Function parameter can handle.
+    :param workspace_index: Workspace index which will be fitted.
+    :param start_x: Start X for the fit
+    :param end_x: End X for the fit
+    :returns: Dataset containing all of Fit's outputs
+    """
+    try:
+        import mantid.simpleapi as mantid
+    except ImportError:
+        raise ImportError(
+            "Mantid Python API was not found, please install Mantid framework "
+            "as detailed in the installation instructions (https://scipp."
+            "readthedocs.io/en/latest/getting-started/installation.html)")
+
+    fit = mantid.Fit(Function=function,
```

Comment:
Also leaking workspaces?

Reply:
Good catch. I did not find a `StoreInADS=False`-like functionality for `Fit`, so I've just resorted to manually calling `DeleteWorkspace` on the 3 outputs. Also added checks in the tests